src/client/main.c

```diff
@@ -3,52 +3,88 @@
 #include <getopt.h>
 #include <stdlib.h>
 
-#include "libbluechi/common/common.h"
+#include "libbluechi/cli/command.h"
 #include "libbluechi/common/opt.h"
 
 #include "client.h"
+#include "method-help.h"
+#include "method-list-units.h"
+#include "method-loglevel.h"
+#include "method-metrics.h"
+#include "method-monitor.h"
+#include "method-status.h"
+#include "method-unit-actions.h"
 
-const struct option options[] = {
+#define OPT_NONE 0u
+#define OPT_HELP 1u << 0u
+#define OPT_FILTER 1u << 1u
+
+int method_version(UNUSED Command *command, UNUSED void *userdata) {
+        printf("bluechictl version %s\n", CONFIG_H_BC_VERSION);
+        return 0;
+}
+
+// TODO: add specific usage functions for each method replacing usage-bluechi,
```

Comment:
NIT: TODOs as inline-comments will basically be forgotten about - could you remove this comment and create an gh issue instead? (I know its kind of the same for issues, but its at least a bit more visible :wink:)

Reply:
Good point :)
Done
https://github.com/eclipse-bluechi/bluechi/issues/691